src/header.jsx

```diff
@@ -115,3 +119,47 @@ const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
         </Select>
     );
 };
+
+const UploadButton = ({ currentDir, files }) => {
+    const ref = useRef();
+
+    const handleClick = () => {
+        ref.current.click();
+    };
+
+    const onUpload = e => {
+        const uploadedFile = e.target.files[0];
+
+        let fileName = uploadedFile.name;
+
+        // eslint-disable-next-line no-loop-func
+        while (files.some(f => f.name === fileName)) {
+            fileName += " (1)";
+        }
+
+        const reader = new FileReader();
+        reader.readAsArrayBuffer(uploadedFile, "UTF-8");
+        reader.onload = readerEvent => {
+            const channel = cockpit.channel({
+                binary: true,
+                payload: "fsreplace1",
+                path: currentDir + fileName
+            });
+
+            const buffer = readerEvent.target.result;
+            channel.send(buffer);
+            channel.control({ command: "done" });
+            channel.close();
+        };
+    };
+
+    return (
+        <>
+            <Button variant="secondary" onClick={handleClick}>Upload</Button>
```

Comment:
I'm not very familiar with that `ref` usage here. What does it do? It seems weird that a click event handler runs `.click()` on itself.

Reply:
Because `<input>` handles the file upload change event, and we want to control it with a patternfly styled button. There might be a better way, I haven't looked deep into it